### PR TITLE
feat: fw-1646 refine compose ui/ux

### PR DESCRIPTION
**Pull request by swkatmask**
https://mask.atlassian.net/browse/fw-1646

### CODE REVIEW

**Comment by swkatmask** [comment]
@guanbinrui @dudu0506 PTAL

File: src/components/ComposeButton/ComposeButtonForMobile.tsx [hunk 1]
```diff
@@ -25,8 +25,8 @@ export function ComposeButtonForMobile() {
     const currentPost = useCurrentVisitingPost();
     const currentChannel = useCurrentVisitingChannel();
 
-    if (!isLogin) return null;
-    if (isPostPage && !isCurrentLogin) return null;
+    // if (!isLogin) return null;
+    // if (isPostPage && !isCurrentLogin) return null;
```
Comment (guanbinrui) [inline src/components/ComposeButton/ComposeButtonForMobile.tsx:29]:
```suggestion
```

File: src/components/Compose/useAddVideo.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        (file: File) => {
+            const maxVideoSize = getPostVideoSizeLimit(availableSources);
+            if (file.size > maxVideoSize) {
+                enqueueErrorMessage(
+                    t`The video "${file.name}" (${formatFileSize(file.size, false)}) exceeds the size limit (${formatFileSize(maxVideoSize, false)}).`,
+                );
+                return false;
+            }
+            updateVideo(createLocalMediaObject(file));
+            return;
```
Comment (guanbinrui) [inline src/components/Compose/useAddVideo.ts:25]:
```suggestion
            return true;
```

File: src/components/Compose/useAddImages.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                }
+                return isValidFileType(file.type);
+            });
+            updateImages((images) => {
+                if (images.length === maxImageCount) return images;
+                return [...images, ...shouldUploadFiles.map((file) => createLocalMediaObject(file))].slice(
+                    0,
+                    maxImageCount,
+                );
+            });
```
Comment (guanbinrui) [inline src/components/Compose/useAddImages.ts:39]:
```suggestion
            });
            return true;
```

File: src/components/Compose/ComposeActions/types.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export interface ActionProps {
```
Comment (guanbinrui) [inline src/components/Compose/ComposeActions/types.ts:1]: Declare it in the component that uses it.

File: src/components/Compose/ComposeActions/ChannelAction.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                        </Popover.Button>
+                        <ChannelSearchPanel onSelected={close} className="max-h-[192px]" />
+                    </>
+                )}
+            </Popover>
+        );
+
+    return (
+        <>
+            <button
```
Comment (guanbinrui) [inline src/components/Compose/ComposeActions/ChannelAction.tsx:48]: Use `<ClickableButton />` stead.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**